frontend/src/metabase/parameters/components/ValuesSourceModal/ValuesSourceModal.unit.spec.tsx

```diff
@@ -255,16 +248,33 @@ describe("ValuesSourceModal", () => {
             name: "Products",
           }),
         ],
-        hasDataAccess: false,
+        hasCollectionAccess: false,
+      });
+
+      expect(
+        screen.getByText("You don't have permissions to do that."),
+      ).toBeInTheDocument();
+    });
+
+    it("should allow searching for a card without access to the root collection (metabase#30355)", async () => {
+      await setup({
+        hasCollectionAccess: false,
       });
 
+      userEvent.click(
+        screen.getByRole("radio", { name: "From another model or question" }),
+      );
+      userEvent.click(
+        screen.getByRole("button", { name: /Pick a model or question…/ }),
```

Comment:
nit: I'd be consistent on using string vs regexp

Reply:
It's not possible easily use string here because of an inner icon that has label